xExtension-WordHighlighter/extension.php

```diff
@@ -0,0 +1,103 @@
+	public bool $enable_in_article = false;
+	public bool $enable_logs = false;
+	public bool $case_sensitive = false;
+	public bool $separate_word_search = false;
+
+	#[\Override]
+	public function init(): void
+	{
+		$this->registerTranslates();
+
+		// register CSS for WordHighlighter:
+		Minz_View::appendStyle($this->getFileUrl('style.css', 'css'));
+
+		Minz_View::appendScript($this->getFileUrl('mark.min.js', 'js'), false, false, false);
+
+		$current_user = Minz_Session::paramString('currentUser');
+
+		$staticPath = join_path($this->getPath(), 'static');
+		$configFileJs = join_path($staticPath, ('config.' . $current_user . '.js'));
+
+		if (file_exists($configFileJs)) {
+			Minz_View::appendScript($this->getFileUrl(('config.' . $current_user . '.js'), 'js'));
+		}
+
+		Minz_View::appendScript($this->getFileUrl('word-highlighter.js', 'js'));
+	}
+
+	#[\Override]
+	public function handleConfigureAction(): void
+	{
+		$this->registerTranslates();
+
+		$current_user = Minz_Session::paramString('currentUser');
+		$staticPath = join_path($this->getPath(), 'static');
+
+		$configFileJson = join_path($staticPath, ('config.' . $current_user . '.json'));
+
+		if (!file_exists($configFileJson) && !is_writable($staticPath)) {
+			$tmpPath = explode(EXTENSIONS_PATH . '/', $staticPath);
+			$this->permission_problem = $tmpPath[1] . '/';
+
+		} elseif (file_exists($configFileJson) && !is_writable($configFileJson)) {
+			$tmpPath = explode(EXTENSIONS_PATH . '/', $configFileJson);
+			$this->permission_problem = $tmpPath[1];
+
+		} elseif (Minz_Request::isPost()) {
+			$configWordList = html_entity_decode(Minz_Request::paramString('words_list'));
+
+			$this->word_highlighter_conf = $configWordList;
+			$this->enable_in_article = (bool) Minz_Request::paramString('enable-in-article');
+			$this->enable_logs = (bool) Minz_Request::paramString('enable_logs');
+			$this->case_sensitive = (bool) Minz_Request::paramString('case_sensitive');
+			$this->separate_word_search = (bool) Minz_Request::paramString('separate_word_search');
+
+			$lineSeparator = ((bool) strpos($configWordList, "\r\n")) ? "\r\n" : "\n";
+			$configObj = [
+				'enable_in_article' => $this->enable_in_article,
+				'enable_logs' => $this->enable_logs,
+				'case_sensitive' => $this->case_sensitive,
+				'separate_word_search' => $this->separate_word_search,
+				'words' => explode($lineSeparator, $configWordList),
+			];
+			$configJson = json_encode($configObj, WordHighlighterExtension::JSON_ENCODE_CONF);
+			file_put_contents(join_path($staticPath, ('config.' . $current_user . '.json')), $configJson . PHP_EOL);
+			file_put_contents(join_path($staticPath, ('config.' . $current_user . '.js')), $this->jsonToJs($configJson) . PHP_EOL);
+		}
+
+		if (file_exists($configFileJson)) {
+			try {
+				$confJson = json_decode(file_get_contents($configFileJson) ?: '', true, 8, JSON_THROW_ON_ERROR);
+				if (json_last_error() !== JSON_ERROR_NONE || !is_array($confJson)) {
```

Comment:
I do not believe `json_last_error() !== JSON_ERROR_NONE` will ever be relevant when you also use `JSON_THROW_ON_ERROR`. I believe the idea is to use one or the other, not both.